packages/@react-spectrum/actionbar/src/ActionBar.tsx

```diff
@@ -82,6 +82,33 @@ const ActionBarInner = React.forwardRef((props: ActionBarInnerProps, ref: DOMRef
     announce(formatMessage('actionsAvailable'));
   }, [formatMessage]);
 
+  let restoreFocusRef = useRef<HTMLElement>(null);
+
+  // When selectedItemCount changes, update the restoreFocusRef for match the last document.activeElement
+  useLayoutEffect(() => {
+    if (selectedItemCount !== 0 && lastCount.current) {
+      restoreFocusRef.current = document.activeElement as HTMLElement;
+    }
+  }, [selectedItemCount]);
```

Comment:
Hmm, so we only update the `restoreFocusRef` when an item selection is made? Ideally IMO it would be best if we could track what was the last focused cell/row in the TableView since the user could shift their focus via arrow keys before tabbing into the ActionBar. 

Been playing around with this locally to see if there is a way to do this. One DOM crawly way perhaps would be to find what element has `tabIndex = 0` in ActionBar container (excluding the ActionBar itself) and manually focus that when the ActionBar closes. The downside is that it is makes an assumption that the component wrapped in the ActionBarContainer is a TableView or at least follows the roving tab index pattern (also is quite DOM crawly which we try to avoid). Haven't been able to figure out a different way unfortunately

Reply:
Actually, could we not get rid of the the useLayoutEffect and the useEffect below and just do
```
  useEffect(() => {
    return () => {
      let walker = getFocusableTreeWalker(domRef.current.parentElement, {tabbable: true});
      let node = walker.nextNode() as HTMLElement;
      requestAnimationFrame(() => {
        node && node.focus()
      });
    }
  }, [domRef]);
```
Or are we anticipating situations where we don't have a SelectableCollection component wrapped within the ActionBar? Should we then instead record what the last document.activeElement was when focus enters the ActionBar and restore focus to that element if it still exist when the ActionBar unmounts?

Reply:
I think I have a solution for this with https://github.com/adobe/react-spectrum/pull/2451/commits/eab13f0bf66288444bd59d01045a3b048d9f1772